src/html5/parser/document.rs

```diff
@@ -185,20 +185,28 @@ impl TreeBuilder for DocumentTaskQueue {
         new_id
     }
 
-    fn create_text(&mut self, content: &str, parent_id: NodeId) {
+    fn create_text(&mut self, content: &str, parent_id: NodeId) -> NodeId {
```

Comment:
Making `create_text` and `create_comment` return a node ID is not mandatory but I did it for convenience (in the tests) and also it makes the tree builder interface more consistent anyways..